model/common/src/icon4py/model/common/grid/icon.py

```diff
@@ -127,11 +127,22 @@ def __eq__(self, other: "IconGrid"):
 
     @utils.chainable
     def with_start_end_indices(
-        self, dim: gtx.Dimension, start_indices: np.ndarray, end_indices: np.ndarray
+        self,
+        dim: gtx.Dimension,
+        start_indices: field_alloc.NDArray,
+        end_indices: field_alloc.NDArray,
     ):
         log.debug(f"Using start_indices {dim} {start_indices}, end_indices {dim} {end_indices}")
-        self._start_indices[dim] = start_indices.astype(gtx.int32)
-        self._end_indices[dim] = end_indices.astype(gtx.int32)
+        self._start_indices[dim] = (
+            start_indices.ndarray.astype(gtx.int32)
+            if hasattr(start_indices, "ndarray")
+            else start_indices.astype(gtx.int32)
```

Comment:
Same as above: Can you drop the if branch and rerun. 

```suggestion
start_indices.astype(gtx.int32)
```